pkg/scheduler/framework/session_plugins.go

```diff
@@ -304,6 +309,50 @@ func (ssn *Session) JobOrderFn(l, r interface{}) bool {
 
 }
 
+// NamespaceOrderEnabled returns the NamespaceOrder is enabled in this session or not
+// judging by length of namespaceOrderFns
+func (ssn *Session) NamespaceOrderEnabled() bool {
+	return len(ssn.namespaceOrderFns) > 0
+}
+
+// NamespaceOrderEnabledInPlugin the NamespaceOrder for this plugin is enabled in this session or not
+func (ssn *Session) NamespaceOrderEnabledInPlugin(pluginName string) bool {
+	for _, tier := range ssn.Tiers {
+		for _, plugin := range tier.Plugins {
+			if plugin.Name != pluginName {
+				continue
+			}
+			return isEnabled(plugin.EnabledNamespaceOrder)
+		}
+	}
+	return false
+}
+
+// NamespaceOrderFn invoke namespaceorder function of the plugins
+func (ssn *Session) NamespaceOrderFn(l, r interface{}) bool {
+	for _, tier := range ssn.Tiers {
+		for _, plugin := range tier.Plugins {
+			if !isEnabled(plugin.EnabledNamespaceOrder) {
+				continue
+			}
+			nof, found := ssn.namespaceOrderFns[plugin.Name]
+			if !found {
+				continue
+			}
+			if j := nof(l, r); j != 0 {
+				return j < 0
+			}
+		}
+	}
+
+	// TODO(lminzhw): if all NamespaceOrderFn treat these two namespace as the same,
+	// we should make the job order have its affect among namespaces.
+	// or just schedule namespace one by one
+	lv := l.(string)
```

Comment:
Do we need to check if this converting failing?

Reply:
this actions following the code pattern in other `CompareFn`, like `taskOrder` or `jobOrder` in `priority` plugin.